src/components/Search/SelectedRevisionsTable/SelectedRevisionsTable.tsx

```diff
@@ -0,0 +1,106 @@
+import { Alignment } from '../../Shared/enums';
+import StyledTableCell from '../../Shared/StyledTableCell';
+import StyledTableRow from '../../Shared/StyledTableRow';
+// eslint-disable-next-line import/no-unresolved
+import './SelectedRevisionsTable.css';
+
+interface TableHeaderDetails {
+  title: string;
+  alignment: Alignment;
+}
+
+const tableHeaderDetails: TableHeaderDetails[] = [
+  { title: 'Project', alignment: 'left' },
+  { title: 'Revision', alignment: 'left' },
+  { title: 'Author', alignment: 'left' },
+  { title: 'Commit message', alignment: 'center' },
+  { title: 'Timestamp', alignment: 'left' },
+];
+
+function SelectedRevisionsTable(props: any) {
+  const { revisions } = props;
+  console.log(revisions);
+  const dispatch = useDispatch();
+
+  if (revisions.length === 0) {
+    return <div className="missingRevisions">Add some revisions</div>;
+  }
+  return (
+    <TableContainer className="layout">
+      <Table style={{ minWidth: 700 }}>
+        <TableHead>
+          <TableRow>
+            <StyledTableCell />
+            {tableHeaderDetails.map((header) => (
+              <StyledTableCell align={header.alignment} className="title">
+                {header.title}
+              </StyledTableCell>
+            ))}
+            <StyledTableCell />
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {revisions.map((row: any, index: number) => (
+            <StyledTableRow key={row.id.toString()}>
+              <StyledTableCell component="th" scope="row">
+                <div className="cellStyle">{index === 0 ? 'BASE' : 'NEW'}</div>
+              </StyledTableCell>
+              <StyledTableCell component="th" scope="row">
+                {row.projectName}
+              </StyledTableCell>
+              <StyledTableCell align="left">{row.revision}</StyledTableCell>
+              <StyledTableCell align="left">{row.author}</StyledTableCell>
+              <StyledTableCell align="center">
+                {row.commitMessage}
+              </StyledTableCell>
+              <StyledTableCell align="left">{row.timeStamp}</StyledTableCell>
+              <StyledTableCell align="right">
+                <IconButton
+                  id="close-button"
+                  onClick={() => dispatch(deleteRevision(row.id))}
+                >
+                  <Close />
+                </IconButton>
+              </StyledTableCell>
+            </StyledTableRow>
+          ))}
+        </TableBody>
+      </Table>
+      <div className="helperText">Maximum 4 revisions.</div>
+    </TableContainer>
+  );
+}
+
+SelectedRevisionsTable.propTypes = {
+  revisions: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.number,
+      projectName: PropTypes.string,
+      revision: PropTypes.string,
+      author: PropTypes.string,
+      commitMessage: PropTypes.string,
+      timeStamp: PropTypes.string,
+    }),
+  ).isRequired,
+};
```

Comment:
Should this be changed with an `interface`?

Reply:
yeah I think it should be..good catch, thankyou.